graylog2-server/src/main/java/org/graylog2/streams/matchers/MatchInput.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.streams.matchers;
+
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.streams.StreamRule;
+
+public class MatchInput implements StreamRuleMatcher {
+
+    @Override
+    public boolean match(Message msg, StreamRule rule) {
+       if(msg.getField(Message.FIELD_GL2_SOURCE_INPUT) == null) {
+           return rule.getInverted();
+       }
+        final String value = msg.getField(Message.FIELD_GL2_SOURCE_INPUT).toString();
+        return rule.getInverted() ^ value.trim().equals(rule.getValue());
```

Comment:
IMO we can do a case-insensitive check in here, as afaics input IDs should be unique regarding the case.